components/developer/netbeans/netbeans.p5m

```diff
@@ -0,0 +1,7952 @@
+#
+# This file and its contents are supplied under the terms of the
+# Common Development and Distribution License ("CDDL"), version 1.0.
+# You may only use this file in accordance with the terms of version
+# 1.0 of the CDDL.
+#
+# A full copy of the text of the CDDL should have accompanied this
+# source.  A copy of the CDDL is also available via the Internet at
+# http://www.illumos.org/license/CDDL.
+#
+
+#
+# Copyright 2017 Jim Klimov
+# Copyright 2013 Andrzej Szeszo (see studio-netbeans.p5m)
+#
+
+<transform file path=.*/bin/.+ -> default mode 0555>
+<transform file path=.*\.so -> default mode 0555>
+<transform file path=.*\.sh -> default mode 0555>
+
+# Setting exec bit on Windows objects allows to share them over kCIFS or NFS
+# and allow Windows guests to run them from the network share according to ACL
+<transform file path=.*\.exe -> default mode 0555>
+<transform file path=.*\.dll -> default mode 0555>
+
+<transform file dir link hardlink path=.* -> default pkg.linted true>
+
+# Do not complain about certain files
+<transform file path=.*/[Ll]inux.+ -> default pkg.depend.bypass-generate .*>
+<transform file path=.*/hpux.+ -> default pkg.depend.bypass-generate .*>
+<transform file path=.*/FreeBSD.+ -> default pkg.depend.bypass-generate .*>
+<transform file path=.*/MacOSX.+ -> default pkg.depend.bypass-generate .*>
+<transform file path=.*/rfs_controller -> default pkg.depend.bypass-generate .*>
+<transform file path=.*/fs_server -> default pkg.depend.bypass-generate .*>
+<transform file path=.*/runant\.py -> default pkg.depend.bypass-generate .*>
+
+# Keep config/tracking files as is, if modified by user (normally these should
+# be changed in homedir and not in installation directory)
+<transform file path=.*/etc/.+ -> default preserve renamenew>
+<transform file path=.*/config/.+ -> default preserve renamenew>
+<transform file path=.*/update_tracking/.+ -> default preserve renamenew>
+<transform file path=.*/update/.+ -> default preserve renamenew>
+<transform file path=.*\.conf -> default preserve renamenew>
+
+set name=pkg.fmri value=pkg:/$(COMPONENT_FMRI)@$(IPS_COMPONENT_VERSION),$(BUILD_VERSION)
+set name=pkg.summary value="$(COMPONENT_SUMMARY)"
+set name=info.classification value=org.opensolaris.category.2008:Development/C value=org.opensolaris.category.2008:Development/C++ value="org.opensolaris.category.2008:Development/High Performance Computing" value="org.opensolaris.category.2008:Development/Integrated Development Environments" value=org.opensolaris.category.2008:Development/Suites
+set name=info.upstream-url value=$(COMPONENT_PROJECT_URL)
+set name=info.source-url value=$(COMPONENT_ARCHIVE_URL)
+set name=org.opensolaris.consolidation value=$(CONSOLIDATION)
+
+license $(COMPONENT_LICENSE_FILE) license='$(COMPONENT_LICENSE)'
+
+# JRE should suffice to run the IDE; JDK would be needed to develop for Java
+#depend type=require fmri=pkg:/runtime/java/openjdk8
```

Comment:
Of course - C, C++, Python/Jython, PERL, PHP... Mostly via plugins (some may be in core distro) so I did not define dependencies from IDE to languages/compilers. Java is certainly needed to run it though :)